### PR TITLE
Add PhoneNumber, DateOfBirth, URL variable types 

### LINKED ISSUE #443: Add PhoneNumber, DateOfBirth, URL variable types
### Add PhoneNumber, DateOfBirth, URL variable types
-----
- Add PhoneNumber Variable Type
- Add DateOfBirth Variable Type 
- Add URL Variable Type
- Add these types to ` featuretools/tests/testing_utils/mock_ds.py`
- Add to `docs/source/api_reference.rst`
- **Two PR Examples of adding new Variable Types #430 and #426**
```python
class PhoneNumber(Variable):
    _dtype_repr = "phone_number"
    _default_pandas_dtype = str

class DateOfBirth(Datetime):
    _dtype_repr = "date_of_birth"
    _default_pandas_dtype = np.datetime64

class URL(Variable):
    _dtype_repr = "url"
    _default_pandas_dtype = str
```

### PR DESCRIPTION
**Pull request by glentennis**
closes #443 

@gsheni wanna take a look? I wasn't exactly sure if I was supposed to make changes to `featuretools/tests/entityset_tests/test_entity.py`, but I took a stab at it anyway.

Also made a couple changes to unrelated files via `make lint-fix`

### CODE REVIEW

File: featuretools/tests/testing_utils/mock_ds.py [hunk 1]
```diff
@@ -70,9 +70,11 @@ def make_ecommerce_files(with_integer_time_index=False, base_path=None, file_loc
         'upgrade_date': customer_times['upgrade_date'],
         'cancel_date': customer_times['cancel_date'],
         'cancel_reason': ["reason_1", "reason_2", "reason_1"],
-        'date_of_birth': customer_times['date_of_birth'],
         'engagement_level': [1, 3, 2],
         'email': ['john.smith@example.com', '', np.nan],
+        'url': ['google.com', 'https://www.featuretools.com/', np.nan],
```
Comment (gsheni) [inline featuretools/tests/testing_utils/mock_ds.py]: URL is property better to put in the `products` dataframe.

File: featuretools/variable_types/variable.py [hunk 2]
```diff
@@ -257,6 +257,24 @@ class EmailAddress(Variable):
     _default_pandas_dtype = str
 
 
+class URL(Variable):
+    """Represents a valid web url"""
```
Comment (gsheni) [inline featuretools/variable_types/variable.py]: We should put more information in this docstring. Saying that having a url with/without `http` and `www` is okay.

File: featuretools/variable_types/variable.py [hunk 3]
```diff
@@ -257,6 +257,24 @@ class EmailAddress(Variable):
...
+    _dtype_repr = "url"
+    _default_pandas_dtype = str
+
+
+class PhoneNumber(Variable):
+    """Represents a valid phone number"""
```
Comment (gsheni) [inline featuretools/variable_types/variable.py]: We should put more information in this docstring. Saying that having a phone number with/without area and country code is okay, as well as a phone number with parentheses.

**Comment by glentennis** [comment]
@gsheni addressed your comments! if this looks good, I can squash and re-push, which I think will take care of the CLA issue

**Comment by gsheni** [comment]
@glentennis you need to agree to the Contributor License Agreement, please follow these steps
screen shot 2019-02-25 at 10 33 12 am
<br/>
Then click here:
screen shot 2019-02-25 at 10 33 03 am

Or try click this link
https://cla-assistant.io/Featuretools/featuretools?pullRequest=447

**Comment by glentennis** [comment]
Hey @gsheni, looks like we're good to go on this!

**Comment by kmax12** [comment]
@glentennis fyi `_dtype_repr` changed to `type_string` in #361

**Comment by glentennis** [comment]
thanks! totally missed that

**Comment by glentennis** [comment]
@gsheni checks passed, can you merge this?

**Pull request merged**